### ISSUE TITLE
Out argument val escape should not contribute to invocation escape analysis

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#64783]
https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoTBlADhGAHmAE8AXAUwD4ACAWRQAoBKGgb0xq5rwOPOo0INALw0AzmQgBjANYQANgoD20mqUoBtACxIAugG5O3AG4QATjXMUyomgHkYFewFcyvGACUbL8zHEejMpuQsxGGNw0CADsVjbhAL6YOPhEGoJ0SCzsxlwe/JS0wmKSMvJKquoCOvrhkQAqAO7Kru6p4kEhEAA0NMG2+em0wGG5UbHAickYgwK0js5uHt5kvv6B/TypBYIQzJgcEdwA9Mc0AOQxQuc0UOK3ShQA5oo0LjBkUAq3tnd9JhRzI1zFAyJQYGNijQACYUABmEBcCjIdW4V1hCKRKMwSSwGAQ2hoTRaS3anQG2yGQl6m1mhXU+wwh0ipwuUOANwI0IuwDsEE51geCmer3en2+yicNAAFhB7oprBBoSR1BQKDAaMDQeDIXZeWIMYjkVMMEA==

```cs
using System;

Span<byte> M1() {
    Span<byte> a = stackalloc byte[42];
    var ret = OneOutSpanReturnsSpan(out a);
    return ret; // unexpected error is reported
}

Span<byte> M2() {
    Span<byte> a = stackalloc byte[42];
    TwoOutSpans(out a, out Span<byte> b);
    return b; // unexpected error is reported
}

Span<byte> OneOutSpanReturnsSpan(out Span<byte> a)
{
    // 'return a' is illegal until it is overwritten
    a = default;
    return default;
}

void TwoOutSpans(out Span<byte> a, out Span<byte> b)
{
    // 'a = b' and 'b = a' are illegal until one has already been written
    a = b = default;
}
```

**Comment by RikkiGibson** [comment]
@jaredpar I didn't find language in the [spec](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-11.0/low-level-struct-improvements.md) which says out arguments do not contribute their safe-to-escape to invocations, but I seem to recall you saying the language is there. Could you please point it out for me?
